spark-commons-test/src/main/scala/za/co/absa/spark/commons/test/SparkTestBase.scala

```diff
@@ -19,7 +19,9 @@ package za.co.absa.spark.commons.test
 import org.apache.spark.sql.SparkSession
 
 trait SparkTestBase {
-  implicit final val spark: SparkSession = initSpark()
+  implicit lazy val spark: SparkSession = initSpark()
```

Comment:
Probably for the better. The original should still work in case the implicit is defined outside of the class extending `SparkTestBase`. But this allows the definition inside as well.
Hopefully not that much taxing on performance. 🤞